File: C++/perfect.cpp
#include <iostream>
using namespace std;

int main() {
    int num, sum;
    sum = 0;
    num = 20'000;
    
    for (int j = num; j >= 1; j--) {
        for ( int i = num - 1; i >= 1; i--) {
            if (num % i == 0) {
                sum += i;
            }
        }
        if (num == sum) {
            cout << "num: " << num << "/ sum:" << sum << endl;
        }
        sum = 0;
        num --;
    }
}

File: rust/perfect.rs
use std::time::{SystemTime, SystemTimeError};

fn main() -> Result<(), SystemTimeError> {
    let mut num = 20_000;
    let mut sum = 0;
    let sys_time = SystemTime::now();
    let mut j = num;
    while  j >= 1 {
         let mut i = num - 1;
        while i >= 1 {
            if num % i == 0 {
                sum += i;
            }
            i -= 1;
        }
        if num == sum
        {
            println!{"num: {num} / sum:{sum} "};
        }
        sum = 0;
        num -= 1;
        j -= 1;
    }
    let elaps = sys_time.elapsed()?;
    println!{"The program executed in {}.{} sec", elaps.as_secs(), elaps. subsec_millis()};
    Ok(())
}
